simulation/agents.py

```diff
@@ -611,8 +611,32 @@ def compute_heat_pump_capacity_kw(self, heat_pump_type: HeatingSystem) -> int:
             )
         )
 
+    def proba_of_becoming_heat_pump_aware_required_to_reach_campaign_target(
+        self, model
+    ) -> float:
+        return (
+            model.campaign_target_heat_pump_awareness - model.heat_pump_awareness
+        ) / (1 - model.heat_pump_awareness)
+
+    def update_heat_pump_awareness(self, model) -> None:
+
+        if InterventionType.HEAT_PUMP_CAMPAIGN in model.interventions:
+            if (
+                model.current_datetime == model.heat_pump_awareness_campaign_date
```

Comment:
Is this hard `==` safe? If user input is say '2028-01-31' will this work?

Maybe a safer one would be model.current_datetime >= model.heat_pump_awareness_campaign_date..

Reply:
I had to change the logging to implement this. See latest commit. 

In summary: we want to say "if the target HP awareness has not been met in the previous timestep (t-1), give a suitable probability of converting agents to becoming HP aware to meet the target at timestep t", however, before the HP awareness was recalculated every timestep (every timestep, HP awareness set to value of zero, then using the line of code in agents.py: `model.households_heat_pump_aware_at_current_step += 1` we count the HP awareness in that timstep). Remember, the simulation loops over each agent every timestep. So if you are on agent num. 3 in the loop, the HP awareness is calculated by looking at how many of the 3 agents are heat pump aware so far, and on timestep 5, we see the HP awareness by  looking at how many of the 5 agents are heat pump aware so far etc. So we need to store the HP awareness at t-1, to access it at t, so we can convert a suitable number of households to become HP aware to meet the target.